Readme.md

```diff
@@ -1,5 +1,154 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set some rules
+
+First you will need to add some rules to the BNF before we can set the rules
+```rust
+NumberList ::= number (sep number)*
+sep ::= " "
+number ::= [0-9]+
+```
+`add_rule_for_names` takes in a 2 arguments. The First one being a `list` or in our case `vec!` of names you want to set a rule for.
+The second argument is the rule you want to set, specifiying how theses rules should be handled. By default the lexer will insert `CullStrategy::LiftChildren` if no argument is found. 
+
+`CullStrategy::None` - Leaves the tokens alone
+
+`CullStrategy::DeleteAll` - Deletes the token and all of its children
+
+`CullStrategy::DeleteChildren` - Deletes the children of the token but not the token itself
+
+`CullStrategy::LiftChildren` - Deletes the token and replaces it with its children in its parent
+
+`CullStrategy::LiftAtMost(usize)` - Delete the token and replace it with its children only if it has N or less children
+
+Once done it should look similar to this (depending on your rules and strategy)
+```rust
+lexer.add_rule_for_names(vec!["sep"], CullStrategy::DeleteAll);
+```
+
+
+# Tokens
+
+Then using res we can match it like this to get our tokenized BNF    
+```rust
+lexer.tokenize(test_input).unwrap()
+```
+This can error, but we're not handling it here since this is just an example
```

Comment:
I think I must have miscommunicated how `.unwrap()` works. It's not wrong to explicitly state this, but it's kind of redundant. Having a `.unwrap()` *implies* that this code can fail, but we're not handling the error here since it's just an example, so we don't need to say it explicitly.